Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -309,4 +317,75 @@ lemma exists_maximal_isMatchingFree [Finite V] (h : G.IsMatchingFree) :
   obtain ⟨Gmax, hGmax⟩ := Finite.exists_le_maximal h
   exact ⟨Gmax, ⟨hGmax.1, ⟨hGmax.2.prop, fun _ h' ↦ hGmax.2.not_prop_of_gt h'⟩⟩⟩
 
+/-- A graph `G` consists of a set of cycles, if each vertex is either isolated or connected to
+exactly two vertices. This is used to create new matchings by taking the `symmDiff` with cycles.
+The definition of `symmDiff` that makes sense is the one for `SimpleGraph`. This is why this
+definition is for graphs, rather than subgraphs.
+-/
+def IsCycles (G : SimpleGraph V) := (∀ v : V, (G.neighborSet v) = ∅ ∨ (G.neighborSet v).ncard = 2)
```

Comment:
I have a hunch that this will be easier to use
```suggestion
def IsCycles (G : SimpleGraph V) := ∀ ⦃v⦄, (G.neighborSet v).Nonempty → (G.neighborSet v).ncard = 2
```